code-challenges.test.js

```diff
@@ -14,19 +14,102 @@
 
 // a) Create a test with expect statements using the variables provided.
 
+// RED 
+describe("codedMessage", () => {
+    it("should convert 'Lackadaisical' to 'L4ck4d41s1c4l'", () => {
+      const secretCodeWord1 = "Lackadaisical"
+      expect(codedMessage(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+    })
+  
+    it("should convert 'Gobbledygook' to 'G0bbl3dyg00k'", () => {
+      const secretCodeWord2 = "Gobbledygook"
+      expect(codedMessage(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+    })
+  
+    it("should convert 'Eccentric' to '3cc3ntr1c'", () => {
+      const secretCodeWord3 = "Eccentric"
+      expect(codedMessage(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+  })
+  
+//   ReferenceError: codedMessage is not defined
+//   Test Suites: 1 failed, 1 total
+//   Tests:       3 failed, 3 total
+//   Snapshots:   0 total
+
 const secretCodeWord1 = "Lackadaisical"
 // Expected output: "L4ck4d41s1c4l"
 const secretCodeWord2 = "Gobbledygook"
 // Expected output: "G0bbl3dyg00k"
 const secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
 
+// GREEN 
+// PSEIDOCODE
+
+// input: a string
+// output: a coded message that converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0. ie: const secretCodeWord1 = "Lackadaisical"
+// Expected output: "L4ck4d41s1c4l"
+
+// PROCESS 
+// create a function called codedMessage that takes a string as an argument. 
+// create a variable to hold the letters/numbers that need to be decoded called codeTable
+// create a variable to for an empty String. this will hold the ouput of your decoded message 
+// us a for loop to iterate over the length of the message 
+// create a comparitve statement to check if the letters in codeTable are in the string passes.
+// if the characters from codeTable are present the replacement character will be added to the string using
+// use toLowerCase to account for input casing and lower case any letters input
+// return the coded message 
+
+const codedMessage = (message) => {
+    const codeTable = {
+      "a": "4",
+      "e": "3",
+      "i": "1",
+      "o": "0"
+    };
+    let codedMessage = ""
+    for (let i = 0; i < message.length; i++) {
+      if (codeTable[message[i].toLowerCase()]) {
+        codedMessage += codeTable[message[i].toLowerCase()]
+      } else {
+        codedMessage += message[i]
+      }
+    }
+    return codedMessage
+  }
```

Comment:
Nice approach! I especially like the use of an object to store the replacement values! Quick note: for best practice, variables should have a different name than the function!